fixtures/pages-workerjs-with-routes-app/tests/index.test.ts

```diff
@@ -1,36 +1,95 @@
 import { fork } from "child_process";
 import * as path from "path";
 import { fetch } from "undici";
-import type { ChildProcess } from "child_process";
 
 describe("Pages Advanced Mode with custom _routes.json", () => {
-	let wranglerProcess: ChildProcess;
-	let ip: string;
-	let port: number;
-	let resolveReadyPromise: (value: unknown) => void;
-	const readyPromise = new Promise((resolve) => {
-		resolveReadyPromise = resolve;
-	});
+	it.concurrent(
+		"renders static pages",
+		async () => {
+			const { ip, port, exit } = await runWranglerPagesDev(
+				"public",
+				"--port=0"
+			);
+			const response = await fetch(`http://${ip}:${port}/`);
+			const text = await response.text();
+			expect(text).toContain(
+				"Bienvenue sur notre projet &#10024; pages-workerjs-with-routes-app!"
+			);
+			await exit();
+		},
+		10000
+	);
 
-	beforeAll(() => {
-		wranglerProcess = fork(
-			path.join("..", "..", "packages", "wrangler", "bin", "wrangler.js"),
-			["pages", "dev", "public", "--port=0"],
-			{
-				stdio: ["inherit", "inherit", "inherit", "ipc"],
-				cwd: path.resolve(__dirname, ".."),
-			}
-		).on("message", (message) => {
-			const parsedMessage = JSON.parse(message.toString());
-			ip = parsedMessage.ip;
-			port = parsedMessage.port;
-			resolveReadyPromise(undefined);
-		});
-	});
+	it.concurrent(
+		"runs our _worker.js",
+		async () => {
+			const { ip, port, exit } = await runWranglerPagesDev(
```

Comment:
You've unnecessarily kicked off another server here (could have just re-used the original one), but no biggie.

Reply:
I'll refactor.

Reply:
I tried using `beforeAll` but occasionally would find that `stop/exit` is `undefined` in the `afterAll` hook.